src/System/Hworker.hs

```diff
@@ -776,6 +809,25 @@ monitor hw =
   forever $ do
     now <- getCurrentTime
 
+    runRedis (hworkerConnection hw) $ do
+      R.zcount (scheduleQueue hw) 0 (utcToDouble now) >>=
+        \case
+          Right n | n > 0 ->
+            withNil' hw $
+              R.eval
+                "local jobs = redis.call('zrangebyscore', KEYS[1], '0', ARGV[1])\n\
+                \redis.call('lpush', KEYS[2], unpack(jobs))\n\
+                \redis.call('zremrangebyscore', KEYS[1], '0', ARGV[1])\n\
+                \return nil"
+                [scheduleQueue hw, jobQueue hw]
+                [B8.pack (show (utcToDouble now))]
+
+          Right _ ->
+            return ()
+
+          Left err ->
+            liftIO $ hwlog hw err
+
```

Comment:
So I wasn't sure if this was at all necessary, but since this monitor is polling pretty frequently and most of the time there won't be anything in the scheduled queue. I figured I'd have it at least do an initial check to see if anything is there before bothering to run the Lua script, since I know that's blocking. But maybe this is pointless?